### PR TITLE
Empty list semantic

### LINKED ISSUE #62: Nil in morganey
I think it would be nice, of Morganey was able to express nil.

Nil (or for example []) could also be a literal for the empty list (#58).

### PR DESCRIPTION
**Pull request by rexim**
Close #62

### CODE REVIEW

**Comment by rexim** [comment]
Alright, everything seems to be compiling.

@ForNeVeR @keddelzz could you please take a look when you get a chance? Thanks!

File: kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala [hunk 1]
```diff
@@ -17,32 +17,28 @@ class ChurchPairConverterSpec extends FlatSpec with Matchers with TestTerms {
...
-    val term = LambdaParser.parse(LambdaParser.term, rawTerm).get
-    val expectedList = "xxy".map(c => lvar(c.toString)).toList
-
-    decodeList(term) should be (expectedList)
+    val inputTerm = pair(x, pair(x, pair(y, zero)))
+    val expectedList = Some("xxy".map(c => lvar(c.toString)).toList)
+    decodeList(inputTerm) should be (expectedList)
   }
 
   "A church list of number" should "be converted to some list of numbers" in {
```
Comment (ForNeVeR) [inline kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala]: "A church list of number**S**"?

File: macros/src/main/scala/me/rexim/morganey/meta/Liftable.scala [hunk 2]
```diff
@@ -4,7 +4,7 @@ import scala.language.higherKinds
 
 import me.rexim.morganey.ast.LambdaTerm
 import me.rexim.morganey.church.ChurchNumberConverter.encodeNumber
-import me.rexim.morganey.church.ChurchPairConverter.{encodeList, encodePair}
+import me.rexim.morganey.church.ChurchPairConverter.{encodeList, encodePair, encodeString}
```
Comment (ForNeVeR) [inline macros/src/main/scala/me/rexim/morganey/meta/Liftable.scala]: I cannot see where're you using `encodeString` in this file.

File: macros/src/test/scala/me/rexim/morganey/meta/InterpolatorSpec.scala [hunk 3]
```diff
@@ -58,7 +58,7 @@ class InterpolatorSpec extends FlatSpec with Matchers with TestTerms {
     val m"${nTwo: Int}" = two
     nTwo should be (2)
 
-    val m"${numbers: List[Int]}" = pair(zero, pair(one, two))
+    val m"${numbers: List[Int]}" = pair(zero, pair(one, pair(two, zero)))
     numbers should be (List(0, 1, 2))
   }
```
Comment (ForNeVeR) [inline macros/src/test/scala/me/rexim/morganey/meta/InterpolatorSpec.scala:67]: It'd be nice to add an explicit test for an empty list here.

File: kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala (hunk 1 above)
Comment (ForNeVeR) [inline kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala]: Also I'm not sure but I think you should capitalize Church at least in string literals like this. [Wikipedia](https://en.wikipedia.org/wiki/Church_encoding) does. So I suggest "A **Church** list of **numbers**".

Comment (rexim) [inline kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala]: done both

File: macros/src/main/scala/me/rexim/morganey/meta/Liftable.scala (hunk 2 above)
Comment (rexim) [inline macros/src/main/scala/me/rexim/morganey/meta/Liftable.scala]: Removed that

File: macros/src/test/scala/me/rexim/morganey/meta/InterpolatorSpec.scala (hunk 3 above)
Comment (rexim) [inline macros/src/test/scala/me/rexim/morganey/meta/InterpolatorSpec.scala:67]: Added a test for that

**Comment by ForNeVeR** [comment]
LGTM :+1:

**Comment by rexim** [comment]
@ForNeVeR ok, thanks for the review! I think it's time to merge that!
